tests/pipeline/test_blaze.py

```diff
@@ -1315,46 +1337,73 @@ def test_deltas(self, asset_info, add_extra_sid):
             name='delta',
             dshape=self.dshape,
         )
-        expected_views = keymap(pd.Timestamp, {
+        expected_views_all_deltas = keymap(pd.Timestamp, {
             '2014-01-02': np.array([[10.0, 11.0, 12.0],
                                     [1.0, 2.0, 3.0]]),
             '2014-01-03': np.array([[11.0, 12.0, 13.0],
                                     [2.0, 3.0, 4.0]]),
             '2014-01-04': np.array([[12.0, 13.0, 14.0],
                                     [12.0, 13.0, 14.0]]),
         })
+        expected_views_novel_deltas = keymap(pd.Timestamp, {
+            '2014-01-02': np.array([[0.0, 1.0, 2.0],
+                                    [1.0, 2.0, 3.0]]),
+            '2014-01-03': np.array([[1.0, 2.0, 3.0],
+                                    [2.0, 3.0, 4.0]]),
+            '2014-01-04': np.array([[2.0, 3.0, 4.0],
+                                    [2.0, 3.0, 4.0]]),
+        })
 
         nassets = len(asset_info)
+
         if nassets == 4:
-            expected_views = valmap(
-                lambda view: np.c_[view, [np.nan, np.nan]],
-                expected_views,
+            expected_views_all_deltas = self._get_fourth_asset_view(
+                expected_views_all_deltas, window_length=2
+            )
+            expected_views_novel_deltas = self._get_fourth_asset_view(
+                expected_views_novel_deltas, window_length=2
             )
         with tmp_asset_finder(equities=asset_info) as finder:
-            expected_output = pd.DataFrame(
-                list(concatv([12] * nassets, [13] * nassets, [14] * nassets)),
-                index=pd.MultiIndex.from_product((
-                    sorted(expected_views.keys()),
-                    finder.retrieve_all(asset_info.index),
-                )),
-                columns=('value',),
+            expected_output_all_deltas = self._get_expected_output(
+                expected_views_all_deltas,
+                [12, 13, 14],
+                finder,
+                asset_info,
+                nassets
             )
-            dates = self.dates
-            dates = dates.insert(len(dates), dates[-1] + timedelta(days=1))
-            self._run_pipeline(
-                expr,
-                deltas,
-                None,
-                expected_views,
-                expected_output,
+            expected_output_novel_deltas = self._get_expected_output(
+                expected_views_novel_deltas,
+                [3, 4, 4],
                 finder,
-                calendar=dates,
-                start=dates[1],
-                end=dates[-1],
-                window_length=2,
-                compute_fn=np.nanmax,
+                asset_info,
+                nassets
             )
 
+            dates = self.dates
+            dates = dates.insert(len(dates), dates[-1] + timedelta(days=1))
+            for apply_deltas_adjustments, expected_views, expected_output in (
```

Comment:
instead of running with and without the novel deltas for every test, can you just add one new test case that does this? Right now we are mixing this into all the cases when that is not the core part under test. This will also slow down the test suite a lot.